src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -713,18 +713,37 @@ public static unsafe double Round(double value, int digits, MidpointRounding mod
 
                 value *= power10;
 
-                if (mode == MidpointRounding.AwayFromZero)
+                if (mode is MidpointRounding.ToEven)
                 {
-                    var fraction = ModF(value, &value);
+                    // Rounds to the nearest value; if the number falls midway,
+                    // it is rounded to the nearest value with an even least significant digit
+                    value = Round(value);
+                }
+                else if (mode is MidpointRounding.AwayFromZero)
+                {
+                    // Rounds to the nearest value; if the number falls midway,
+                    // it is rounded to the nearest value above (for positive numbers) or below (for negative numbers)
+                    double fraction = ModF(value, &value);
 
                     if (Abs(fraction) >= 0.5)
                     {
                         value += Sign(fraction);
                     }
                 }
-                else
+                else if (mode is MidpointRounding.ToZero)
                 {
-                    value = Round(value);
+                    // Directed rounding: Round toward zero, to nearest value above (positive numbers) or below (negative numbers)
```

Comment:
I think the comment needs fixing, we won't be rounding above for positive numbers, but rather the nearest exact or the nearest just below (closer to zero)